PdCore/src/main/java/org/puredata/android/service/PdService.java

```diff
@@ -241,8 +241,15 @@ private Notification makeNotification(Intent intent, int icon, String title, Str
 				notificationManager.createNotificationChannel(channel);
 			}
 		}
-
-		PendingIntent pi = PendingIntent.getActivity(getApplicationContext(), 0, intent, 0);
+		
+            	PendingIntent pi = null;
+            	if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                	pi = PendingIntent.getActivity(getApplicationContext(), 0, intent, PendingIntent.FLAG_UPDATE_CURRENT | PendingIntent.FLAG_IMMUTABLE);
+            	}
+            	else {
+                	pi = PendingIntent.getActivity(getApplicationContext(), 0, intent, PendingIntent.FLAG_UPDATE_CURRENT);
```

Comment:
What is the reason for changing the value that was previously used (`0`) to `PendingIntent.FLAG_UPDATE_CURRENT`?
```suggestion
                	pi = PendingIntent.getActivity(getApplicationContext(), 0, intent, PendingIntent.FLAG_IMMUTABLE);
            	}
            	else {
                	pi = PendingIntent.getActivity(getApplicationContext(), 0, intent, 0);
```

Reply:
In my experience it doesn't change much, but we can keep 0 to be more like the original code.
https://stackoverflow.com/questions/50128188/what-are-different-between-pendingintent-flag-update-current-and-0-in-android

Reply:
Let's keep `0` then for now.

Reply:
This might be clearer with a helper to compose the flag(s)